internal/querynodev2/tasks/scheduler.go

```diff
@@ -151,7 +155,23 @@ func (s *Scheduler) processAll(ctx context.Context) {
 }
 
 func (s *Scheduler) process(t Task) {
+	s.cond.L.Lock()
+	for s.runningThreadNum >= s.pool.Cap() {
+		s.cond.Wait()
+	}
+	s.runningThreadNum += t.Weight()
+	s.cond.L.Unlock()
+
 	s.pool.Submit(func() (any, error) {
+		defer func() {
+			s.cond.L.Lock()
+			defer s.cond.L.Unlock()
+			s.runningThreadNum -= t.Weight()
+			if s.runningThreadNum < s.pool.Cap() {
+				s.cond.Signal()
```

Comment:
Signal notifies only one goroutine which is waiting. Might introduce some corner case that some search request is never notified.

```suggestion
				s.cond.Broadcast()
```

Reply:
Here only 1 goroutine is scheduling the task